spyql/processor.py

```diff
@@ -283,16 +313,20 @@ def eval_clause(self, clause, clause_exprs, mode="eval"):
             )
 
     # main
-    def go(self, output_file, output_options):
+    def go(self, output_file, output_options, user_query_vars = {}):
         output_handler = OutputHandler.make_handler(self.prs)
         writer = Writer.make_writer(self.prs["to"], output_file, output_options)
         output_handler.set_writer(writer)
-        nrows_in = self._go(output_handler)
+        nrows_in = self._go(output_handler, user_query_vars)
         output_handler.finish()
         spyql.log.user_info("#rows  in", nrows_in)
         spyql.log.user_info("#rows out", output_handler.rows_written)
 
-    def _go(self, output_handler):
+        # TODO: @yashbonde fix the condition
+        if hasattr(writer, "out"):
+            return writer.out
```

Comment:
This is nice. 
One note is that if we decide to return a result object in the future, we might also include statistics like the #rows in/out, among other potential helpful stuff